src/core/public/overlays/overlay_service.ts

```diff
@@ -75,19 +61,17 @@ export class OverlayService {
 export interface OverlayStart {
   /** {@link OverlayBannersStart} */
   banners: OverlayBannersStart;
-  openFlyout: (
-    flyoutChildren: React.ReactNode,
-    flyoutProps?: {
-      closeButtonAriaLabel?: string;
-      'data-test-subj'?: string;
-    }
-  ) => OverlayRef;
-  openModal: (
-    modalChildren: React.ReactNode,
-    modalProps?: {
-      className?: string;
-      closeButtonAriaLabel?: string;
-      'data-test-subj'?: string;
-    }
-  ) => OverlayRef;
+  /** {@link OverlayModalStart} */
+  modals: OverlayModalStart;
+  /** {@link OverlayFlyoutStart} */
+  flyouts: OverlayFlyoutStart;
+
+  /**
+   * @deprecated Use {@link OverlayStart.flyouts} instead
+   */
+  openFlyout: OverlayFlyoutStart['open'];
+  /**
+   * @deprecated Use {@link OverlayStart.modals} instead
+   */
+  openModal: OverlayModalStart['open'];
```

Comment:
The PR is big already, I only deprecated these methods instead of totally removing them. This can be done is as a next step

Reply:
Would really like to see these removed or them internally use the new services. Or will these be removed in a subsequent PR?

Reply:
They are already only function bindings to the new services. Jot sure about the deprecation policy in Kibana, but if we keep them, the idea would be to definitely removed them in next major I guess.

However seems like we might rollback on this, and keeping them instead of the new `overlay.(modal|flyout).open` approach the PR is currently implementing